src/gpu/pca/pca.cu

```diff
@@ -0,0 +1,51 @@
+#include "../data/matrix.cuh"
+#include "../device/device_context.cuh"
+#include "pca.h"
+#include "../tsvd/tsvd.h"
+
+namespace pca
+{
+
+/**
+ * Conduct PCA on a matrix
+ *
+ * @param _X
+ * @param _Q
+ * @param _w
+ * @param _U
+ * @param _explained_variance
+ * @param _explained_variance_ratio
+ * @param _param
+ */
+void pca(const double* _X, double* _Q, double* _w, double* _U, double* _explained_variance, double* _explained_variance_ratio, params _param) {
+	try {
+
+		//Take in X matrix and allocate for X^TX
+		tsvd::Matrix<float>X(_param.X_m, _param.X_n);
+		X.copy(_X);
+
+		tsvd::Matrix<float>XtX(_param.X_n, _param.X_n);
+
+		//create context
+		tsvd::DeviceContext context;
+
+        tsvd::normalize_columns(X, context);
```

Comment:
This is what I anticipated you using. However, there is a `cub` version. Have you looked into that?

Reply:
@navdeep-G no, will check if this one turns out to be a bottleneck